### DISCUSSION TITLE
Post request to thanos is not working from grafana

### DISCUSSION THREAD

**Discussion by jatindersandhu** [thanos-io/thanos#4337] [Questions & Answers]
Hi,
    We have openshift cluster. We are tanent in cluster want to see POD level matrics. Openshift team give us thanos url , it is working fine for GET request but give ERROR Bad Gateway on POST request.  We are using thanos tanent port 9092. Is thanos support POST level Request
Here is version
sh-4.4$ thanos --version
thanos, version 0.17.2 (branch: rhaos-4.7-rhel-8, revision: e08776699e995facd3260ebfec4c29c4d3bd4943)
  build user:       root@3d4eeeb5b403
  build date:       20210521-04:48:20
  go version:       go1.15.7
  platform:         linux/amd64

**Comment by jatindersandhu** [comment]
any update please

**Comment by GiedriusS** [comment]
Yes, it does https://github.com/thanos-io/thanos/blob/47a001547370abbc5a44a9a1eb3d68786e23e6d8/pkg/api/query/v1.go#L162-L190. Please take a look at the logs to see what is wrong. I hope this helps.

**Reply by jatindersandhu** [reply]
Thanks for confirmation. We also test on main thanos port it is working but the problem here is that we have multi tenants environment. For that we use prom_label_proxy and create new thanos tenant port  (9092) , on which only GET is working not POST.
